quipucords/api/reports/view.py

```diff
@@ -41,25 +41,27 @@
 @authentication_classes(auth_classes)
 @permission_classes(perm_classes)
 @renderer_classes((ReportsGzipRenderer,))
-def reports(request, pk=None):
+def reports(request, report_id=None):
     """Lookup and return reports."""
     reports_dict = {}
     mask_report = request.query_params.get("mask", False)
-    if pk is not None:
-        if not is_int(pk):
+    if report_id is not None:
+        if not is_int(report_id):
             error = {"report_id": [_(messages.COMMON_ID_INV)]}
             raise ValidationError(error)
```

Comment:
Every time I see this pattern I want to remove it because:

1. there's no URL mapped to this view without pk/report_id
2. weird HTTP semantics: how NOT FOUND is a 400? LOL

Don't need to remove it, tho. Just wanted to point how weird this is

Reply:
Oh yes, I wanted to refactor this when I saw it too, but I had to make myself stop or I'd be rewriting everything all the time. 😂 Glad we agree, though!